File: src/types/knowledge-base.ts
// Knowledge Base types
export interface KnowledgeBase {
  id: string
  name: string
  type: string // "document", "url", "file", "folder"
  content?: string
  url?: string
  metadata?: string
  created_at: string
  updated_at: string
}

export interface CreateKnowledgeBaseRequest {
  name: string
  type: string
  content?: string
  url?: string
  metadata?: string
}

File: src-tauri/src/models/knowledge_base.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use ts_rs::TS;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow, TS)]
#[ts(export)]
pub struct KnowledgeBase {
    pub id: String,
    pub name: String,
    pub r#type: String,
    #[ts(optional)]
    pub content: Option<String>,
    #[ts(optional)]
    pub url: Option<String>,
    #[ts(optional)]
    pub metadata: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, TS)]
#[ts(export)]
pub struct CreateKnowledgeBaseRequest {
    pub name: String,
    pub r#type: String,
    #[ts(optional)]
    pub content: Option<String>,
    #[ts(optional)]
    pub url: Option<String>,
    #[ts(optional)]
    pub metadata: Option<String>,
}
